### DISCUSSION TITLE
ECS for axmol?

### DISCUSSION THREAD

**Discussion by rudiHammad** [axmolengine/axmol#1235] [General]
Hello,
I was wondering if someone has tried to write an ECS for axmol. I saw a couple of links in cocos2dx forum
refering to some github repositories like entt, or wizardfu ecs. Has anyone tried to implement them in axmol?
thanks

**Comment by aismann** [comment]
@rudiHammad 
Why not using one of the cocos2dx/ECS example as start point?

**Comment by rudiHammad** [comment]
Are you referring to this one [https://github.com/NatWeiss/EntityFu](https://github.com/NatWeiss/EntityFu) ? 
I guess that the one from wizardfu is fine, but since it is from 2016 I wasn't sure about it.
that is the only cocos2dx/ECS I know. Is there anyone one?

**Reply by antn9x** [reply]
Hi @rudiHammad , I tried to use axmol with entityX. And success to combined them. I really want to share my work if you interested.

**Reply by rudiHammad** [reply]
hi antn9x. I am actually writing my own ECS, and I am starting to understand it quite well . So see how others integrate existing ECS with axmol would help a lot if that's okey.

**Reply by antn9x** [reply]
Writing another library is not wise. For study, it's ok. but when it's grow many things to do.

**Reply by rudiHammad** [reply]
I agree with you. I am writing ECS to learn it. So far I succeed and have already a couple of system working. So it is in a pretty good shape and learned a lot just by trying to do my own. But yeah, I agree with you. I'll see how it evolves and might switch to entityx or another depending on how it goes.

**Comment by rudiHammad** [comment]
By the way, when en it comes to things like loading levels, UIs and other similar elements that are not part of  the game loop, is ECS necessary there or is it okey to apply OOP. For instance, maybe a factory pattern to create the game levels. 
I've read trying to that going 100% ECS is a mistake an that we should combine both worlds.

**Reply by antn9x** [reply]
ECS should be implement for all elements. It's not just optimze for game loop, also best practice for ui, When your system complete, it work perfect when we just need add more data to render UI.

**Reply by rudiHammad** [reply]
I see. 
cool, thank you.
